packages/desktop/electron/lib/errorPreload.js

```diff
@@ -0,0 +1,19 @@
+const { contextBridge, ipcRenderer } = require('electron')
+const { version } = require('../../package.json')
+
+contextBridge.exposeInMainWorld('error', {
+    getData: (callback) => {
+        ipcRenderer.invoke('error-data').then(data => {
+            callback({
+                iconPath: './assets/logos/firefly_logo.svg',
+                version,
+                diagnostics: data.diagnostics.map((d) => `${d.label.replace('popups.diagnostics.', '')}: ${d.value}`).join('\r\n'),
+                errorType: data.errorType,
+                error: data.error
+            })
+        })
+    },
```

Comment:
I think you can avoid using a callback here:

```suggestion
    getData: () => {
        return ipcRenderer.invoke('error-data').then((data) => {
            return {
                iconPath: './assets/logos/firefly_logo.svg',
                version,
                diagnostics: data.diagnostics.map((d) => `${d.label.replace('popups.diagnostics.', '')}: ${d.value}`).join('\r\n'),
                errorType: data.errorType,
                error: data.error
            }
        })
    },
```

Reply:
Agreed, you would never guess I copied the code from the about window ;)